api/integrations_aws_gov_cloud.go

```diff
@@ -0,0 +1,119 @@
+//
+// Author:: Darren Murray (<darren.murray@lacework.net>)
+// Copyright:: Copyright 2021, Lacework Inc.
+// License:: Apache License, Version 2.0
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+//
+
+package api
+
+// NewAwsGovCloudIntegration returns an instance of AwsGovCloudIntegration with the provided
+// integration type, name and data.
+//
+// Basic usage: Initialize a new AwsGovCloudIntegration struct, then
+//              use the new instance to do CRUD operations
+//
+//   client, err := api.NewClient("account")
+//   if err != nil {
+//     return err
+//   }
+//
+//   aws := api.NewAwsGovCloudIntegration("foo",
+//     api.AwsGovCloudIntegration,
+//     api.AwsGovCloudIntegrationData{
+//       Credentials: api.AwsGovCloudCreds {
+//         AccountID: "553453453",
+//         AccessKeyID: "AWS123abcAccessKeyID",
+//         SecretAccessKey: "AWS123abc123abcSecretAccessKey0000000000",
+//       },
+//     },
+//   )
+//
+//   client.Integrations.CreateAwsGovCloudCfg(aws)
+//
+func NewAwsGovCloudCfgIntegration(name string, iType integrationType, data AwsGovCloudIntegrationData) AwsGovCloudIntegration {
```

Comment:
uhmmm, this is confusing, why are we passing an integration type if we are specifically saying that we are creating an `AWS_US_GOV_CFG` integration? I think this was a copy-paste from the main `NewAwsIntegration()` function, actually, did you try to use it instead?

Could we do something like:
```go
   client, err := api.NewClient("account")
   if err != nil {
     return err
   }

   aws := api.NewAwsIntegration("foo",
     api.AwsGovCloudCfgIntegration,
     api.AwsIntegrationData{
       GovCloudCredentials: api.AwsGovCloudCreds {},
     },
   )

   client.Integrations.CreateAws(aws)
```

Reply:
I'm not sure Salim, would AwsIntegrationData then have to accept GovCloudCredentials type?